Controller.cpp

```diff
@@ -523,6 +591,28 @@ namespace Procon {
 		return sendCommand(0x10, buf);
 	}
 
+	Controller::exchangeArray Controller::getSpiData(uint32_t offset, const uint8_t readLen)
+	{
+		std::array<uint8_t, 0x100> buf;
+		while (true)
+		{
+			buf.fill(0);
+			*(uint32_t*)(buf.data()) = offset;
+			*(uint8_t*)(buf.data() + 4) = readLen;
```

Comment:
No reason to obfuscate this, just do
```c++
buf[4] = readLen;
```

Reply:
I think we can do that. But I also thought it looks strange to use another method of access here.